src/WP_CLI/CommandWithUpgrade.php

```diff
@@ -500,6 +500,11 @@ static function ( $result ) {
 	// phpcs:ignore PSR2.Methods.MethodDeclaration.Underscore -- Whitelisting to provide backward compatibility to classes possibly extending this class.
 	protected function _list( $_, $assoc_args ) {
 
+		// If `--force-check` flag is present, delete the upgrade transient.
+		if ( true === (bool) Utils\get_flag_value( $assoc_args, 'force-check', false ) ) {
+			delete_site_transient( $this->upgrade_transient );
+		}
+
```

Comment:
What should happen if someone uses `--force-check` and `--skip-update-check` at the same time?

Reply:
Hi @danielbachhuber,

That's a good question. As currently written, `--force-check` is toothless if `--skip-update-check` is also passed to the command.

Here are a couple of proposals for resolving this matter (assuming we're solely talking about plugins, as an example):

1. Let the code for both functions execute—delete the plugin transient and skip the call to `wp_update_plugins()`—and issue a warning: "Plugin updates were requested to be force-checked and were not checked due to the --skip-update-check argument."
2. Make it so that the two options are mutually exclusive: you can run either one or the other. Running both results in an error: "Plugins cannot be both force-checked and skipped. Choose one."
3. Let one command always "over-rule" the other. For example, we might decide that `--skip-update-check` should always prevail if both are present. It can proceed quietly (and not delete the plugin transient) or issue a warning ("Plugin updates were not force-checked due to the --skip-update-check argument."). On the other hand, we could decide that `--force-check` always over-rules `--skip-update-check`.

I think I'm partial to option 2 but can be convinced of choosing one of the other options (or alternatives unmentioned above).

Thoughts?

Reply:
@baizmandesign Option 2 works!

Reply:
Sounds good, @danielbachhuber! I'll get to work on implementing the error and the theme version of the flag.